tools/src/test/scala/scala/scalanative/build/TargetTripleTest.scala

```diff
@@ -0,0 +1,37 @@
+package scala.scalanative.build
+
+import org.scalatest._
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+class TargetTripleTest extends AnyFlatSpec with Matchers {
+
+  val cases = List(
+    "aarch64-unknown-linux-gnu" ->
+      TargetTriple("aarch64", "unknown", "linux", "gnu"),
```

Comment:
I discovered all these test cases by running `clang --version` on different machines and CI runners.